run.go

```diff
@@ -139,38 +135,102 @@ var commandInspect = cli.Command{
 	Action: actionInspect,
 }
 
-func guessType(filename string) string {
-	switch {
-	case strings.Contains(filename, "://"):
-		return TypeURL
-	case strings.HasSuffix(filename, ".js"):
-		return TypeJS
-	default:
-		return ""
+func looksLikeURL(str []byte) bool {
+	// Try to parse string as URL
+	u, err := url.Parse(strings.TrimSpace(string(str)))
+	if err != nil || u.Scheme == "" {
+		return false
 	}
+	return true
 }
 
-func makeRunner(filename, t string) (lib.Runner, error) {
-	if t == TypeAuto {
-		t = guessType(filename)
+func getSrcData(arg, t string) (*lib.SourceData, error) {
+	srcdata := []byte("")
+	srctype := t
+	filename := arg
+	var err error
```

Comment:
Avoid doing this, I can see why you're doing it (easy composite assignment), but it's more idiomatic to assign to a temporary in a nested scope, check the error, then assign the inner to the outer.